tests/components/recorder/test_purge.py

```diff
@@ -138,7 +138,7 @@ def test_purge_method(hass, hass_recorder):
             hass.data[DATA_INSTANCE].block_till_done()
             wait_recording_done(hass)
             assert (
-                mock_logger.debug.mock_calls[5][1][0]
+                mock_logger.debug.mock_calls[6][1][0]
```

Comment:
Should we extend a test case a bit to cover this? I could think of two cases:
1. Two states, both about to be deleted. `old_state_id` of state_1  is `state_2`. -> Tests that purge still works.
2. Two states, state_1 -> purge; state_2 -> keep. `old_state_id` of state_2 set to `state_1`. -> Tests that `old_state_id` is correctly set to `null`.

Reply:
Yes, thats a good idea.